### ISSUE TITLE
Advanced Vehicle Damage - Multiple Hits and Catastrophic Kills 

### ISSUE THREAD

**Issue by Fullback14** [acemod/ACE3#8629]
I absolutely love the new advanced vehicle damage feature, but there are certain points where I feel it merits improvement.

Right now, tanks will take multiple consecutive effective hits from tank-calibre guns and ATGMs and remain "alive" and intact in-game terms (intended behaviour I know) and even move and retaliate for a short time before the damage registers, but with the kind of massive damage sometimes received one expects a catastrophic kill, maybe even, especially from WARPAC tanks, turret popping (which is also missing this version, I understand it has issues?)   Even one ATGM can be sufficient like in: https://www.youtube.com/watch?v=MWRAyHD47rw. I know there is already a existing consideration for this, since GBUs still annihilate targets as expected, but if I'm not misunderstanding the code, it's currently based on single hit damage and is set high enough that it only ever applies to bombs (which I have no problem with), and doesn't take into account multiple hits.

Proposal: Add a (configurable?) chance for singular penetrating hits from appropriately large-caliber or lethal-effect rounds (ie. ATGMs) that are below the damage threshold currently required to guarantee total destruction, to cause a catastrophic kill, dependent on the component(s) hit. (effectively setdamage 1 state, maybe sometimes how the old cookoff used to work for tanks) Add a much higher (hard-coded?) chance for multiple consecutive penetrating hits to outright K-kill vehicles, and if possible turret popping for tanks. (though push comes to shove we can probably live without that just fine, since that's realistically always only gonna be for T-series tanks and Merkavas) Re-add HE-type rounds' capability to outright destroy light vehicles and cars ala WT's hull break, which also used to be a thing, to give players more incentive to switch ammo types against these and save on their tank-killers.  

Alternatively, though I'm unsure if this is even possible, you could make the vehicle treated as destroyed by AVD (enough damage to components, overall damage, etc) registered as !alive by the game. Again, I don't think this is feasible, or it might be more trouble than what it's worth, but I'm no expert.

Though this does have value for MP and SP both in terms of immersion and gameplay, I have SP gameplay in mind for this, where players  without the benefit of a Zeus have to tick objectives that require vehicles to be destroyed but still want to benefit from the enhanced features and effects of AVD in the same mission. As always, your hard work with this mod is highly appreciated!

TL;DR: https://www.youtube.com/watch?v=uW-R7RDVJI8

**Comment by pterolatypus** [comment]
Personally I'd rather not have more random chances lying around.
My suggestion would be to just make the instakill threshold ([here?](https://github.com/acemod/ACE3/blob/3c1c0812390fe475e2266432d2cb7b42cdf42c77/addons/vehicle_damage/functions/fnc_processHit.sqf#L36)) a setting, and then store total damage from multiple hits somewhere so that it can push the vehicle over this threshold. It would also be neat to have cookoff and burning add damage to this counter, so a sufficiently-damaged vehicle can be visibly wrecked after a big cookoff.

I appreciate that the point of the module is basically to remove the "destroyed" state entirely, but this causes problems for scripted missions so a compromise would be nice.

**Comment by LarsAsprah** [comment]
> Personally I'd rather not have more random chances lying around. My suggestion would be to just make the instakill threshold ([here?](https://github.com/acemod/ACE3/blob/3c1c0812390fe475e2266432d2cb7b42cdf42c77/addons/vehicle_damage/functions/fnc_processHit.sqf#L36)) a setting, and then store total damage from multiple hits somewhere so that it can push the vehicle over this threshold. It would also be neat to have cookoff and burning add damage to this counter, so a sufficiently-damaged vehicle can be visibly wrecked after a big cookoff.
> 
> I appreciate that the point of the module is basically to remove the "destroyed" state entirely, but this causes problems for scripted missions so a compromise would be nice.

If you're gonna make the threshold a setting, it would be good for there to be presets (like 6 = tank rounds, 8 = atgm or something) for people unfamiliar with the values.

**Comment by TheCandianVendingMachine** [comment]
I don't think it would be possible to determine tank round vs ATGM. The best I believe that can happen is
a) overall damage
b) overall damage for HE, AP, HEAT, Tandem HEAT

**Comment by Mike-MF** [comment]
Closing due to age, inactivity or already solved. May be re-opened if the issue is still present, even better just open a new up-to-date Issue.
